cmd/gitopsUpdateDeployment.go

```diff
@@ -0,0 +1,158 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/docker"
+	gitUtil "github.com/SAP/jenkins-library/pkg/git"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/go-git/go-git/v5/plumbing"
+	"io"
+	"path/filepath"
+)
+
+type gitopsGitUtils interface {
+	CommitSingleFile(filePath, commitMessage string, worktree gitUtil.UtilsWorkTree) (plumbing.Hash, error)
+	PushChangesToRepository(username, password string, repository gitUtil.UtilsRepository) error
+	PlainClone(username, password, serverURL, directory string) (gitUtil.UtilsRepository, error)
+	ChangeBranch(branchName string, worktree gitUtil.UtilsWorkTree) error
+	GetWorktree(repository gitUtil.UtilsRepository) (gitUtil.UtilsWorkTree, error)
+}
+
+type gitopsFileUtils interface {
+	TempDir(dir, pattern string) (name string, err error)
+	RemoveAll(path string) error
+}
+
+type gitopsExecRunner interface {
+	RunExecutable(executable string, params ...string) error
+	Stdout(out io.Writer)
+	Stderr(err io.Writer)
+}
+
+var gitUtilities gitopsGitUtils = gitUtil.TheGitUtils{}
```

Comment:
this may clash with future steps since it will be available in the `cmd` package

Reply:
```suggestion
var gitopsUpdateDeploymentGitUtilities gitopsGitUtils = gitUtil.TheGitUtils{}
```